### PR TITLE
[ALLUXIO-1322][WIP] Prepared classes for MissingCtor rule in Checkstyle

### PR DESCRIPTION
**Pull request by HorizonNet**
https://alluxio.atlassian.net/browse/ALLUXIO-1322

Prepared some classes for the upcoming MissingCtor rule in Checkstyle. It checks that classes (except abstract ones) define a constructor and don't rely on the default one. The rule will look like the following

<module name="MissingCtor"/>

This ticket is WIP and needs some additional PRs.

### CODE REVIEW

File: core/client/src/main/java/alluxio/client/RemoteBlockReader.java [hunk 1]
```diff
@@ -31,6 +31,10 @@
    * The factory for the {@link RemoteBlockReader}.
    */
   class Factory {
+
+    // prevent instantiation
+    private Factory() {}
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/client/RemoteBlockReader.java]: `private Factory() {} // prevent instantiation`

File: core/client/src/main/java/alluxio/client/RemoteBlockWriter.java [hunk 2]
```diff
@@ -30,6 +30,10 @@
    * The factory for the {@link RemoteBlockWriter}.
    */
   class Factory {
+
+    // prevent instantiation
+    private Factory() {}
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/client/RemoteBlockWriter.java]: ditto

File: core/client/src/main/java/alluxio/client/netty/NettyClient.java [hunk 3]
```diff
@@ -57,6 +57,9 @@
   public static final long TIMEOUT_MS =
       CONF.getInt(Constants.USER_NETWORK_NETTY_TIMEOUT_MS);
 
+  // prevent instantiation
+  private NettyClient() {}
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/client/netty/NettyClient.java]: ditto

File: core/common/src/main/java/alluxio/security/login/LoginModuleConfigurationUtils.java [hunk 4]
```diff
@@ -25,6 +25,9 @@
   /** Class name of Principal according to different OS type. */
   public static final String OS_PRINCIPAL_CLASS_NAME;
 
+  // prevent instantiation
+  private LoginModuleConfigurationUtils() {}
```
Comment (jsimsa) [inline core/common/src/main/java/alluxio/security/login/LoginModuleConfigurationUtils.java]: ditto

File: examples/src/main/java/alluxio/examples/MultiMount.java [hunk 5]
```diff
@@ -34,6 +34,9 @@
  */
 public final class MultiMount {
 
+  // prevent instantiation
+  private MultiMount() {}
```
Comment (jsimsa) [inline examples/src/main/java/alluxio/examples/MultiMount.java]: ditto

File: examples/src/main/java/alluxio/examples/Performance.java [hunk 6]
```diff
@@ -61,6 +61,9 @@
   private static int sBaseFileNumber = 0;
   private static boolean sAlluxioStreamingRead = false;
 
+  // prevent instantiation
+  private Performance() {}
```
Comment (jsimsa) [inline examples/src/main/java/alluxio/examples/Performance.java]: ditto

**Comment by jsimsa** [comment]
@HorizonNet thanks, left some minor comments

**Comment by HorizonNet** [comment]
@jsimsa Thanks for the comments. I updated the PR accordingly.

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**